### PR TITLE
Unset PETSC_DIR during installation

### LINKED ISSUE #2767: Unable to build PETSc, PETSC_DIR already set
`configure` fails immediately if the `PETSC_DIR` environment variable is already set. Spack should unset this.
```
===============================================================================
             Configuring PETSc to compile on your system                       
===============================================================================
TESTING: configureExternalPackagesDir from config.framework(/scratch/ajstewart/spack-stage/spack-stage-sFRYcv/petsc-3.7.4/config/BuildSystem/config/framework.py:834)
TESTING: configureDebuggers from config.utilities.debuggers(/scratch/ajstewart/spack-stage/spack-stage-sFRYcv/petsc-3.7.4/config/BuildSystem/config/utilities/debuggers.py:22)
TESTING: configureGit from config.sourceControl(/scratch/ajstewart/spack-stage/spack-stage-sFRYcv/petsc-3.7.4/config/BuildSystem/config/sourceControl.py:24)
TESTING: configureMercurial from config.sourceControl(/scratch/ajstewart/spack-stage/spack-stage-sFRYcv/petsc-3.7.4/config/BuildSystem/config/sourceControl.py:35)
TESTING: configureCLanguage from PETSc.options.languages(/scratch/ajstewart/spack-stage/spack-stage-sFRYcv/petsc-3.7.4/config/PETSc/options/languages.py:27)
TESTING: configureDirectories from PETSc.options.petscdir(/scratch/ajstewart/spack-stage/spack-stage-sFRYcv/petsc-3.7.4/config/PETSc/options/petscdir.py:23)
*******************************************************************************
         UNABLE to CONFIGURE with GIVEN OPTIONS    (see configure.log for details):
-------------------------------------------------------------------------------
The environmental variable PETSC_DIR /home/ajstewart/petsc MUST be the current directory /scratch/ajstewart/spack-stage/spack-stage-sFRYcv/petsc-3.7.4
*******************************************************************************
```

### PR DESCRIPTION
**Pull request by adamjstewart**
Fixes #2767.

### CODE REVIEW

**Comment by adamjstewart** [comment]
@tgamblin I can do that. Let me just test it first.

File: var/spack/repos/builtin/packages/petsc/package.py [hunk 1]
```diff
@@ -218,6 +218,16 @@ def install(self, spec, prefix):
...
                         '-pc_hypre_type', 'boomeramg')
 
+    def setup_environment(self, spack_env, run_env):
+        # configure fails if these env vars are set outside of Spack
+        spack_env.unset('PETSC_DIR')
+        spack_env.unset('PETSC_ARCH')
+
+        # Set PETSC_DIR in the module file
+        run_env.set('PETSC_DIR', self.prefix)
+        run_env.unset('PETSC_DIR', self.prefix)
```
Comment (davydden) [inline var/spack/repos/builtin/packages/petsc/package.py]: PETSC_ARCH

Comment (davydden) [inline var/spack/repos/builtin/packages/petsc/package.py]: without second argument

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/petsc/package.py]: Can't believe I missed that, thanks!

**Comment by tgamblin** [comment]
@BarrySmith: merging this FYI, in case it impacts you.  I believe it implements what you asked for.

**Pull request merged**